src/Microsoft.AspNetCore.Http.Abstractions/Extensions/UsePathBaseMiddleware.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Builder.Extensions
+{
+    /// <summary>
+    /// Represents a middleware that extracts the specified path base from request path and postpend it to the request path base.
+    /// </summary>
+    public class UsePathBaseMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly PathString _pathBase;
+
+        /// <summary>
+        /// Creates a new instace of <see cref="UsePathBaseMiddleware"/>.
+        /// </summary>
+        /// <param name="next">The delegate representing the next middleware in the request pipeline.</param>
+        /// <param name="pathBase">The path base to extract.</param>
+        public UsePathBaseMiddleware(RequestDelegate next, PathString pathBase)
+        {
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(next));
+            }
+            if (!pathBase.HasValue)
+            {
+                throw new ArgumentException($"{nameof(next)} cannot be null or empty.");
+            }
+
+            _next = next;
+            _pathBase = pathBase;
+        }
+
+        /// <summary>
+        /// Executes the middleware.
+        /// </summary>
+        /// <param name="context">The <see cref="HttpContext"/> for the current request.</param>
+        /// <returns>A task that represents the execution of this middleware.</returns>
+        public async Task Invoke(HttpContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var originalPath = context.Request.Path;
+            PathString remainingPath;
+
+            if (originalPath.StartsWithSegments(_pathBase, out remainingPath))
+            {
+                var originalPathBase = context.Request.PathBase;
+                context.Request.Path = remainingPath;
+                context.Request.PathBase = originalPathBase.Add(originalPath.Value.Substring(0, _pathBase.Value.Length));
```

Comment:
Is there a possibility to combine this line with line __53__ because they essentially just did a string split.